### ISSUE TITLE
Incomplete diagnostic message

### ISSUE THREAD

**Issue by hunterhogan** [microsoft/pyright#11174]
Again, misleading or ambiguous diagnostic messages are a bug in a type checker: what is the value of a diagnostic message that has less information than an Exception thrown by the interpreter?

```python
from cytoolz.itertoolz import concat as concatenate
from itertools import repeat

aa = concatenate((0,), repeat(tuple([-magnitudeOfAddends])))
```

```json
[{
	"resource": "/c:/apps/mapFolding/mapFolding/_e/analysisPython/theExcluderBeast.py",
	"owner": "Pylance",
	"code": {
		"value": "reportCallIssue",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pylance-release/blob/main/docs/diagnostics/reportCallIssue.md",
			"scheme": "https",
			"authority": "github.com"
		}
	},
	"severity": 8,
	"message": "Expected 1 positional argument",
	"source": "Pylance",
	"startLineNumber": 344,
	"startColumn": 25,
	"endLineNumber": 344,
	"endColumn": 61,
	"origin": "extHost1"
}]
```

As often happens, due to the 1) ambiguous and 2) misleading diagnostic message, I spent far too long deconstructing this statement trying to find the problem. Simply using SVO would fix this problem and many other writing problems in the diagnostic messages.

It is hubris to think that because y'all know how to write code, you know how to write.

**Comment by rchiodo** [comment]
Thanks for the issue. Do you have a suggestion for what you want the message to be? `Expected 1 positional argument` seems pretty unambiguous to me.

What did the runtime show? I can't determine that without having the imported packages available.

**Comment by hunterhogan** [comment]
> Do you have a suggestion for what you want the message to be? ``

SVO, Subject Verb Object, because as you know, English does not encode the subject by ~reflecting~ inflecting the verb--unlike nearly every other European language. "Expected 1 positional argument" has an unambiguous verb and object, but who or what subject expected the argument?

In my short statement, at least three subjects may have been expecting the argument (`concatenate`, `repeat`, and `tuple`), and without the S in SVO, the identity of the subject that is disappointed by my failure to provide 1 positional argument is ambiguous.

Based on my literal thousands of hours of writing and editing training at law school with the best legal writing program in the US, if not the world, based on my experience as a teacher, and based on my many years of investigating the issue, a non-professional writer will get the most ROI by simply forcing themselves to always write with SVO unless they can articulate a great reason to not use SVO.

Just SVO. It is that simple. It is not necessarily an _easy_ change, but the goal is simple: SVO, always.

**Comment by hunterhogan** [comment]
> Thanks for the issue. Do you have a suggestion for what you want the message to be? `Expected 1 positional argument` seems pretty unambiguous to me.
> 
> What did the runtime show? I can't determine that without having the imported packages available.

@rchiodo
Dude, I expended time and effort answering _your_ question because _you_, Rich Chiodo, asked the question. You had earned my trust: I believed that if I answered a question _you_ asked, that _you_ would respond. I am shocked that you punted this without any comment. 

And by coincidence, another team at [Microsoft has finally decided that my claims have merit,](https://github.com/microsoft/vscode-pull-request-github/pull/8295#issuecomment-3667545825) even though they handled it wrongfully.

**Comment by rchiodo** [comment]
Sorry but this error comes from Pyright. It's up to Pyright to decide whether or not to change the error.
